plugins/tcpinfo/tcpinfo.cc

```diff
@@ -345,6 +357,41 @@ TSPluginInit(int argc, const char *argv[])
     case 'h':
       hooks = parse_hook_list(optarg);
       break;
+    case 'e':
+      i = strtoul(optarg, &endptr, 10);
+      if (*endptr != '\0' || i > 3) {
+        TSError("[tcpinfo invalid rolling-enabled argument, using default of %d, valid values are 0-3", config->rolling_enabled);
+      } else {
+        config->rolling_enabled = i;
+      }
+      break;
+    case 'H':
+      i = strtoul(optarg, &endptr, 10);
+      if (*endptr != '\0' || i > 23) {
+        TSError("[tcpinfo invalid rolling-offset-hr argument, using default of %d, valid values are 0-23",
```

Comment:
Help the operator out by saying what the invalid value is:

```C
TSError("[tcpinfo] invalid argument `%s` for --rolling-offset-hr", optarg)
```

I think that if there is an invalid option you should just bail out without doing anything. It is the operator's responsibility to read the docs and get it right.